### ISSUE TITLE
VOoM (paste): invalid clipboard--first line is not a headline

### ISSUE THREAD

**Issue by pinggit** [vim-voom/VOoM#6]
I constantly got the below errors when I work on the node tree:

when dd to remove a node

```
VOoM: error setting clipboard  
```

when pp to paste it somewhere else

```
VOoM (paste): invalid clipboard--first line is not a headline
```

have to workaround this via moving node (__ and ^^), this seems always work, but not convenient sometime if the source and target place is too far.

I'm with the current git clone version of VOom and Vim 7.4, what could be the issue?

**Comment by vim-voom** [comment]
On 11/26/15, pinggit <notifications@github.com> wrote:
> I constantly got the below errors when I work on the node tree:
>
> when dd to remove a node
> ```
> VOoM: error setting clipboard
> ```
> when pp to paste it somewhere else
> ```
> VOoM (paste): invalid clipboard--first line is not a headline
> ```
>
> have to workaround this via moving node (__ and ^^), this seems always work,
> but not convenient sometime if the source and target place is too far.
>
> I'm with the current git clone version of VOom and Vim 7.4, what could be
> the issue?


It's clear that yy fails to put lines in the clipboard register but I
have no idea why.
Please provide the following information:
	your operating system
	exact Vim version (:version) and how Vim was installed
	markup mode that you are using with VOoM
	a minimal test file that shows this behavior

Also, try the following commands:
	:echo has('clipboard')

	:let @+="aaa\nbbb"
	:echo @+

Do these commands run without errors?

Also, please provide the output of command `Voominfo all`, it has
useful debugging info.
Place the cursor in a VOoM buffer. Run commands:
	:Voomlog
	:Voominfo all
Then copy the output from the __PyLog__ buffer and paste it here or
email it to me.
In case you do not want to share some private data in the output,
things like home dir name and paths, remove them or blank them with *.
I am especially interested in seeing the value of _VOoM.CLIP .

**Comment by pinggit** [comment]
hi Vlad:

here is the vim version:

:version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Jan  2 2014 19:39:47)
Included patches: 1-52
Modified by pkg-vim-maintainers@lists.alioth.debian.org
Compiled by buildd@
Huge version with GTK2-GNOME GUI.  Features included (+) or not (-):
+acl             +cmdline_hist    +dnd             -footer
 +lispindent      -mouse_jsbterm   +perl            +scrollbind      +tcl
          +viminfo         -xterm_save
+arabic          +cmdline_info    -ebcdic          +fork()
 +listcmds        +mouse_netterm   +persistent_undo +signs
+terminfo        +vreplace        +xpm
+autocmd         +comments        +emacs_tags      +gettext
+localmap        +mouse_sgr       +postscript      +smartindent
+termresponse    +wildignore
+balloon_eval    +conceal         +eval            -hangul_input    +lua
          -mouse_sysmouse  +printer         -sniff           +textobjects
  +wildmenu
+browse          +cryptv          +ex_extra        +iconv           +menu
         +mouse_urxvt     +profile         +startuptime     +title
  +windows
++builtin_terms  +cscope          +extra_search    +insert_expand
+mksession       +mouse_xterm     +python          +statusline
 +toolbar         +writebackup
+byte_offset     +cursorbind      +farsi           +jumplist
 +modify_fname    +multi_byte      -python3         -sun_workshop
 +user_commands   +X11
+cindent         +cursorshape     +file_in_path    +keymap          +mouse
          +multi_lang      +quickfix        +syntax          +vertsplit
  -xfontset
+clientserver    +dialog_con_gui  +find_in_path    +langmap
+mouseshape      -mzscheme        +reltime         +tag_binary
 +virtualedit     +xim
+clipboard       +diff            +float           +libcall
+mouse_dec       +netbeans_intg   +rightleft       +tag_old_static  +visual
         +xsmp_interact
+cmdline_compl   +digraphs        +folding         +linebreak
+mouse_gpm       +path_extra      +ruby            -tag_any_white
+visualextra     +xterm_clipboard
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread
-I/usr/include/gtk-2.0 -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include
-I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr
/include/gdk-pixbuf-2.0 -I/usr/include/pango-1.0
-I/usr/include/gio-unix-2.0/ -I/usr/include/freetype2
-I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include
-I/usr/include/pix
man-1 -I/usr/include/libpng12 -I/usr/include/harfbuzz   -pthread -DORBIT2=1
-D_REENTRANT -I/usr/include/libgnomeui-2.0 -I/usr/include/libart-2.0
-I/usr/include/gconf/2 -I/usr/include/gnome-k
eyring-1 -I/usr/include/libgnome-2.0 -I/usr/include/libbonoboui-2.0
-I/usr/include/libgnomecanvas-2.0 -I/usr/include/gtk-2.0
-I/usr/include/gdk-pixbuf-2.0 -I/usr/include/gnome-vfs-2.0 -I/usr
/lib/x86_64-linux-gnu/gnome-vfs-2.0/include -I/usr/include/dbus-1.0
-I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/glib-2.0
-I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/i
nclude/orbit-2.0 -I/usr/include/libbonobo-2.0
-I/usr/include/bonobo-activation-2.0 -I/usr/include/libxml2
-I/usr/include/pango-1.0 -I/usr/include/gail-1.0 -I/usr/include/harfbuzz
-I/usr/incl
ude/freetype2 -I/usr/include/atk-1.0
-I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/cairo
-I/usr/include/gio-unix-2.0/ -I/usr/include/pixman-1
-I/usr/include/libpng12     -g -O2
-fstack-protector --param=ssp-buffer-size=4 -Wformat
-Werror=format-security -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
-I/usr/include/tcl8.6  -D_REENTRANT=1  -D_THREAD_SAFE=1  -D_LARGEFILE64
_SOURCE=1
Linking: gcc   -L. -Wl,-Bsymbolic-functions -Wl,-z,relro -rdynamic
-Wl,-export-dynamic -Wl,-E  -Wl,-Bsymbolic-functions -Wl,-z,relro
-Wl,--as-needed -o vim   -lgtk-x11-2.0 -lgdk-x11-2.0 -lat
k-1.0 -lgio-2.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lcairo
-lpango-1.0 -lfontconfig -lgobject-2.0 -lglib-2.0 -lfreetype
-lgnomeui-2 -lSM -lICE -lbonoboui-2 -lgnomevfs-2 -lg
nomecanvas-2 -lgnome-2 -lpopt -lbonobo-2 -lbonobo-activation -lORBit-2
-lart_lgpl_2 -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0
-lpangocairo-1.0 -lgdk_pixbuf-2.0 -lcairo -
lpango-1.0 -lfontconfig -lfreetype -lgconf-2 -lgthread-2.0 -lgmodule-2.0
-lgobject-2.0 -lglib-2.0   -lSM -lICE -lXpm -lXt -lX11 -lXdmcp -lSM -lICE
 -lm -ltinfo -lnsl  -lselinux  -lacl -lattr
 -lgpm -ldl  -L/usr/lib -llua5.2 -Wl,-E  -fstack-protector -L/usr/local/lib
 -L/usr/lib/perl/5.18/CORE -lperl -ldl -lm -lpthread -lcrypt
-L/usr/lib/python2.7/config-x86_64-linux-gnu -lpython
2.7 -lpthread -ldl -lutil -lm -Xlinker -export-dynamic -Wl,-O1
-Wl,-Bsymbolic-functions  -L/usr/lib/x86_64-linux-gnu -ltcl8.6 -ldl -lz
-lpthread -lieee -lm -lruby-1.9.1 -lpthread -lrt -ldl -
lcrypt -lm  -L/usr/lib

:echo has('clipboard')
1

vim is installed in ubuntu with apt-get install vim

:let @+="aaa\nbbb"
:echo @+
aaa
bbb
Press ENTER or type command to continue

:Voominfo all
---------- CURRENT VOoM OUTLINE ------------------
 No Name [/home/ping], b50, asciidoc
Body buffer 50, Tree buffer 51
markup mode: asciidoc
[/home/ping/.vim/bundle/VOoM/autoload/voom/voom_mode_asciidoc.pyc]
---------- VOoM INTERNALS ------------------------
_VOoM.FT_MODES = {'md': 'markdown', 'markdown': 'markdown', 'asciidoc':
'asciidoc', 'mkd': 'markdown', 'asciidoc2': 'asciidoc', 'vimwiki':
'vimwiki'}
_VOoM.MODE = ''
_VOoM.CLIP =  '+'
_VOoM.AAMLEFT =  0
_VOoM:           /home/ping/.vim/bundle/VOoM/autoload/voom/voom_vim.pyc
s:script_path = '/home/ping/.vim/bundle/VOoM/autoload/voom.vim'
s:script_dir  = '/home/ping/.vim/bundle/VOoM/autoload'
s:voom_dir    = '/home/ping/.vim/bundle/VOoM/autoload/voom'
g:voom_did_load_plugin = 'v5.1'
s:voom_did_init = 'v5.1'
s:voom_logbnr = 0
s:verify      = 0
g:voom_verify_oop = 1
s:voom_trees  = {'2': 1, '34': 33, '51': 50, '49': 3}
s:voom_bodies = {'1': {'tree': 2, 'mmode': 'asciidoc', 'MTYPE': 1, 'tick_':
30508, 'snLn': 2, 'tick': 30508}, '33': {'tree': 34, 'mmode': 'asciidoc',
'MTYPE': 1, 'tick_': 60, 'snLn': 2, 'tic
k': 62}, '3': {'tree': 49, 'mmode': 'asciidoc', 'MTYPE': 1, 'tick_': 2869,
'snLn': 268, 'tick': 2869}, '50': {'tree': 51, 'mmode': 'asciidoc',
'MTYPE': 1, 'tick_': 39, 'snLn': 2, 'tick': 39}
}

:Voomlog
   1 Python Log buffer ...$
 ~

**Comment by pinggit** [comment]
one thing I noticed, it looks I don't have the + and * register...
not sure if this is the problem? Voom is working based on that?
thanks.

--- Registers ---
""   aaa^Jbbb
"0   aaa^Jbbb
"1   ^J
"2   ^J
"3   ====^J
"4   [NOTE]^J====^JBuilding VMX basedon `SR-IOV` req
"5   == prepare for the installation^J
"6   == progress^J^J.DONE:^J^J* key features involve
"7   About This doc :^J^J* records some of my self-l
"8   This doc recorded some pieces of the knowledge
"9   Juniper MX86(VMX: Virtual MX) is a virtual vers
"a   ping@ubuntu:~$ df -h ^JFilesystem      Size  Us
"o   #login account {{{1}}}^Jset unixpass {~8N\vV.v}
"p   ^J^J::::::::::::::::::::::::^J日期：公元2015年1
"q   [source,bash]^J----^J
"s
"u   ugg:<80>ku^Myyya:w^M
"w   0f/D0jkj
"-   tips
".   s<80>kb
":   Voominfo all
"%   temp-transfer2/asciidoctest.txt
"/   kernel
"=   <SNR>44_SelectCompletion(1)
Press ENTER or type command to continue

**Comment by vim-voom** [comment]
On 12/4/15, pinggit <notifications@github.com> wrote:
> one thing I noticed, it looks I don't have the + and * register...
> not sure if this is the problem? Voom is working based on that?
> thanks.

Yes, VOoM by default uses the + register for yy/dd/pp commands when
has('clipboard') is 1.
Your Vim does have the + register since you can set it and get it with
	:let @+="aaa\nbbb"
	:echo @+

This could have something to do with peculiarities of Linux clipboard handling.

Do you run GUI Vim (gvim) or Vim in terminal?
If terminal, do you have this problem in gvim as well?
What is option 'clipboard' set to? Try resetting it to default (:set cb&vim).
Also try resetting 'guioptions'.

Since the "+ register is problematic, you can tell VOoM to use any "a-z"
register instead. Add to .vimrc and restart Vim:
	let g:voom_clipboard_register = "o"
For details see
	:help g:voom_clipboard_register
Hopefully this will enable yy/dd/pp commands. The drawback is that you will not
be able to cut/copy/paste outlines between different instances of Vim.

**Comment by vim-voom** [comment]
This may be caused by illegal bytes. I found I can reproduce this while investigating Vim issues #1053.
Encoding is utf-8.

    :tabnew
    :0put =printf('aaa%c {{{1',0xFF)
    :Voom

Then "yy" on headline with illegal byte gives error "VOoM: error setting clipboard".
This is Vim Python bug. And it is much worse with Python 3. There is nothing I can do about it. Closing. Let me know if you have more information.
